prov/util/src/util_shm.c

```diff
@@ -62,6 +62,22 @@ static void smr_peer_addr_init(struct smr_addr *peer)
 	peer->addr = FI_ADDR_UNSPEC;
 }
 
+void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
+{
+	struct iovec local_iov, remote_iov;
+	int ret;
+
```

Comment:
We could also add a check here to see if the peer has already checked this. ie
```
   if (peer_smr->cma_cap != -1) {
        smr->cma_cap = peer_smr->cma_cap;
        return;
   }
```

Reply:
Thanks for your suggestion. Updated.